clang/tools/include-mapping/cppreference_parser.py

```diff
@@ -143,32 +165,33 @@ def _GetSymbols(pool, root_dir, index_page_name, namespace, variants_to_accept):
     with open(index_page_path, "r") as f:
         # Read each symbol page in parallel.
         results = []  # (symbol_name, promise of [header...])
+        symbols_by_page = collections.defaultdict(list)
```

Comment:
On my machine it took about 3 minutes for the generator to run, so I tried to optimize it by grouping symbols that are defined on the same page (now it takes 2 minutes).
If the diff is too large or complex, I can move the commit with optimizations to a separate PR.